modules/ggl-http/src/gghttp_util.c

```diff
@@ -47,6 +54,169 @@ static GglError translate_curl_code(CURLcode code) {
     }
 }
 
+static bool can_retry(CURLcode code, CurlData *data) {
+    switch (code) {
+    // If OK, then inspect HTTP status code.
+    case CURLE_OK:
+        break;
+
+    case CURLE_OPERATION_TIMEDOUT:
+    case CURLE_COULDNT_CONNECT:
+    case CURLE_SSL_CONNECT_ERROR:
+    case CURLE_GOT_NOTHING:
+    case CURLE_SEND_ERROR:
+    case CURLE_RECV_ERROR:
+    case CURLE_PARTIAL_FILE:
+    case CURLE_AGAIN:
+        return true;
+
+    default:
+        return false;
+    }
+
+    long http_status_code = 0;
+    curl_easy_getinfo(data->curl, CURLINFO_HTTP_CODE, &http_status_code);
+
+    switch (http_status_code) {
+    case 400: // Generic client error
+    case 408: // Request timeout
+    // TODO: 429 can contain a retry-after header.
+    // This should be used as the backoff.
```

Comment:
if retry-after is like 2 days, we dont want to block that long. Probably would need to cap the accepted value.